source/fab/source_tree.py

```diff
@@ -26,7 +28,13 @@ def __init__(self, extension_map: Mapping[str, Analyser]):
     def visit(self, candidate: Path):
         try:
             analyser = self._extension_map[candidate.suffix]
-            analyser.analyse(candidate)
+            reader: TextReader = FileTextReader(candidate)
+            hasher: TextReaderAdler32 = TextReaderAdler32(reader)
+            analyser.analyse(hasher)
+            for _ in hasher.line_by_line():
+                pass  # Make sure we've read the whole file.
```

Comment:
Only if there is an analyser. Where there is then this loop will not run as the whole file will already have been seen. However if something else is happening it's possible it could give up after it's found what it is interested in. In that case you ned to ensure the rest of the file is pumped through the hasher.

Reply:
Right that makes sense - thanks!